### ISSUE TITLE
Issues with Neopixel and Pi 3B LEDs work but white constant

### ISSUE THREAD

**Issue by Ryuiku** [scottlawsonbc/audio-reactive-led-strip#242]
https://github.com/naztronaut/dancyPi-audio-reactive-led/issues/5#issuecomment-544147078

_Originally posted by @NickSutton in https://github.com/scottlawsonbc/audio-reactive-led-strip/issues/241#issuecomment-544147435_

Hi,

I went through the setup on this from both the original by scottlawsonbc and @joeybab3's repos but i cant seem to get it to run fully right. when i run visualization.py it responds to USB audio on the Pi but the LEDs that aren't responding to music are lit fully white instead of off. regular strandtest.py from rpi_ws281x appears to give me the right colors for the color wipes, but I'm unsure if the theater chase is working right because i feel like some LEDs are supposed to be off instead of fully lit. 

From issue #241 referenced above, I tried changing the strip_type in /rpi_ws281x/python/neopixel.py to `strip_type=ws.SK6812_STRIP_RGBW` from `ws.WS2811_STRIP_RGB` but this just caused random colors to be displayed when running the strandtest.py, SK6812_strandtest.py, and the visualization.py. i also tried to use the specific branch for rpi_ws281x for SK6812 that was mentioned but that was giving me errors trying to run scons on it all (if thats what i need, i can give specifics on that). i can post pictures of connections to rule out improper wiring in a bit. Any help is very appreciated.

Hardware:
-Raspberry Pi 3 Model B (no +)
-Adafruit Neopixel, link: https://www.adafruit.com/product/1461?length=1
-5V 4A power supply.
-Logic Level Converter link: https://www.velleman.eu/products/view/?id=435578

**Comment by Ryuiku** [comment]
![75B63576-FB83-4EFD-AA9A-62AF3765C0A2](https://user-images.githubusercontent.com/56772704/67151611-15692180-f28e-11e9-9818-f9454c7ba477.jpeg)
![926BD173-4A65-4FA7-9FC6-EF65875F6756](https://user-images.githubusercontent.com/56772704/67151612-15692180-f28e-11e9-8366-879db5a43d67.jpeg)

For reference, my setup. Yellow is data, oranges are ground and red is 5v.

**Comment by NickSutton** [comment]
Hey,

I’m not using pin1 to the LEDs, only the external power supply, but apparently it’s important to use the ground as you have.

You’ve linked an RGB (not RGBW) product, which I’m using so you shouldn’t make the changes I made, which are to support the white LED in the strip, which it doesn’t look like yours has?

Did any of the examples in the rpi_ws281x folder work for you?

**Comment by Ryuiku** [comment]
Yeah, the examples from rpi_ws281x kinda worked. The regular strandtest.py worked for the color wipes, but the theater chase didn’t look right. It seemed like when the leds were supposed to be off they were lit fully white.

**Comment by akameswaran** [comment]
Is the level logic inverter set properly?  I had the same thing happen when I had that setting wrong, ie set it to invert when I didn't need to.
